pkg/util/querylimits/propagation.go

```diff
@@ -0,0 +1,80 @@
+package querylimits
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/prometheus/common/model"
+)
+
+// Context key type used to avoid collisions
+type key int
+
+const (
+	queryLimitsContextKey key = 1
+
+	HTTPHeaderQueryLimitsKey = "X-Loki-Query-Limits"
+)
+
+// NOTE: we use custom `model.Duration` instead of standard `time.Duration` because,
+// to support user-friendly duration format (e.g: "1h30m45s") in JSON value.
+type QueryLimits struct {
+	MaxQueryLength   model.Duration `json:"maxQueryLength"`
+	MaxQueryLookback model.Duration `json:"maxQueryLookback"`
+	// this should be changed to max bytes
+	MaxEntriesLimitPerQuery int      `json:"maxEntriesLimitPerQuery"`
```

Comment:
Why is that? I think they both are meaningful. If I'm not mistaken the entry limit is used for log queries.

Reply:
I misunderstood an earlier discussion, thinking we were just using MaxEntries as a placeholder for the future MaxBytes limit